packages/babel-plugin-transform-destructuring/src/index.js

```diff
@@ -57,6 +57,29 @@ export default declare((api, options) => {
     return false;
   }
 
+  /**
+   * Test whether a Node is an expression that evaluates to String.
+   *
+   * Heuristic allowed to give false negatives, but never false positives.
+   */
+  function evaluatesToString(node) {
```

Comment:
Can you add some tests for these checks? Like `var { ["foo" + x]: y } = obj`

Reply:
Sure. I was mostly concerned with avoiding `toPropertyKey` on plain strings and template string expressions; the other cases were kind of afterthought, but I should cover those :)

Reply:
Added some tests for the heuristic.